sdk/python/lib/pulumi/automation/_local_workspace.py

```diff
@@ -442,6 +442,34 @@ def list_stacks(self) -> List[StackSummary]:
             stack_list.append(stack)
         return stack_list
 
+    def list_all_stacks(self) -> List[StackSummary]:
+        result = self._run_pulumi_cmd_sync(["stack", "ls", "--json", "--all"])
+        json_list = json.loads(result.stdout)
+        stack_list: List[StackSummary] = []
+        for stack_json in json_list:
+            stack = StackSummary(
+                name=stack_json["name"],
+                current=stack_json["current"],
+                update_in_progress=(
+                    stack_json["updateInProgress"]
+                    if "updateInProgress" in stack_json
+                    else None
+                ),
+                last_update=(
+                    datetime.strptime(stack_json["lastUpdate"], _DATETIME_FORMAT)
+                    if "lastUpdate" in stack_json
+                    else None
+                ),
+                resource_count=(
+                    stack_json["resourceCount"]
+                    if "resourceCount" in stack_json
+                    else None
+                ),
+                url=stack_json["url"] if "url" in stack_json else None,
+            )
+            stack_list.append(stack)
+        return stack_list
```

Comment:
Again, consider a private shared `def _list_stacks(self, all: bool)` helper with the implementation that can be shared between `list_stacks` and `list_all_stacks`.